SYCL/ESIMD/api/functional/functions/functions_select_2d.hpp

```diff
@@ -0,0 +1,286 @@
+    static_assert(Width > SizeX);
+    static_assert(Height > SizeY);
+    static_assert(Height > OffsetY);
+    static_assert(Width > OffsetX);
+
+    constexpr size_t value_for_increase_ref_data_for_change = 50;
+    static_assert(std::numeric_limits<signed char>::max() >
+                      value_for_increase_ref_data_for_change + NumElems,
+                  "Value that used for increase ref data for fill plus simd "
+                  "size should be less than char max value.");
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> initial_ref_data(NumElems, allocator);
+    shared_vector<DataT> ref_data_for_change(SizeY * SizeX, allocator);
+
+    std::iota(initial_ref_data.begin(), initial_ref_data.end(), 1);
+    // We should have different values in the first reference data and in the
+    // second reference data.
+    std::iota(ref_data_for_change.begin(), ref_data_for_change.end(),
+              initial_ref_data.back() + value_for_increase_ref_data_for_change);
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *init_ref_ptr = initial_ref_data.data();
+      DataT *ref_data_for_change_ptr = ref_data_for_change.data();
+      DataT *const out_ptr = result.data();
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, SizeYT, StrideYT, SizeXT, StrideXT,
+                 SimdSplitCoeffT, OffsetXT, OffsetYT>>([=]() SYCL_ESIMD_KERNEL {
+        TestCaseT::template call_esimd_function<DataT, NumElems, SizeY, StrideY,
+                                                SizeX, StrideX, Height, Width>(
+            init_ref_ptr, ref_data_for_change_ptr, out_ptr, OffsetX, OffsetY);
+      });
+    });
+    queue.wait_and_throw();
+
+    std::vector<size_t> selected_indexes;
+    // Collect the indexess that has been selected.
+    for (int i = 0; i < SizeY; ++i) {
+      for (int j = 0; j < SizeX; ++j) {
+        selected_indexes.push_back(OffsetY * Width + i * Width * StrideY +
+                                   j * StrideX + OffsetX);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto selected_indexses_ptr = selected_indexes.begin();
+
+    for (int i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been selected and changed.
+      if (i < *selected_indexses_ptr) {
+        const DataT &retrieved = result[i];
+        const DataT &expected = initial_ref_data[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+      } else {
+        const DataT &retrieved = result[i];
+        const DataT expected = ref_data_for_change.front();
+        ref_data_for_change.erase(ref_data_for_change.begin());
```

Comment:
Do we need to modify `ref_data_for_change` in verification loop? Maybe it's more efficient just to use an other iterator over the `ref_data_for_change`?

Reply:
the using an iterator on `ref_data_for_change` has been added in this commit: https://github.com/intel/llvm-test-suite/pull/908/commits/dcc5bfa24cc297b36642f2d9ea854fa86a8c266f